src/chat/chatReducers.js

```diff
@@ -56,7 +56,7 @@ export default (state: ChatState = initialState, action: Action) => {
       const messages = state.messages[key] || [];
 
       const newMessages = action.replaceExisting
-        ? action.messages
+        ? action.messages.map(item => messages.find(msg => msg.id === item.id) || item)
```

Comment:
Have a look at the `getStreamsById` selector.
We were considering `getMessagesById` which we haven't have implemented yet. That might be worth doing for perf reasons, for these iterations (there are similar iterations elsewhere).

Consider this here, but also can be implemented as another PR, don't let this slow you down for this one.